src/tensors/linalg.jl

```diff
@@ -299,7 +299,7 @@ function LinearAlgebra.rank(
     r = 0 * dim(first(allunits(sectortype(t))))
     dim(t) == 0 && return r
     S = LinearAlgebra.svdvals(t)
-    tol = max(atol, rtol * maximum(first, values(S)))
+    tol = max(atol, rtol * maximum(S.data))
```

Comment:
```suggestion
    tol = max(atol, rtol * maximum(parent(S)))
```
minor style change to avoid depending on fields